libraries/AP_BattMonitor/AP_BattMonitor_config.h

```diff
@@ -66,6 +66,10 @@
 #define AP_BATTERY_SYNTHETIC_CURRENT_ENABLED AP_BATTERY_BACKEND_DEFAULT_ENABLED
 #endif
 
+#ifndef AP_BATTERY_ACS37800_ENABLED
+#define AP_BATTERY_ACS37800_ENABLED (AP_BATTERY_BACKEND_DEFAULT_ENABLED && (BOARD_FLASH_SIZE > 500))
```

Comment:
```suggestion
#define AP_BATTERY_ACS37800_ENABLED AP_BATTERY_BACKEND_DEFAULT_ENABLED
```

Please also move this towards the top so we keep the backends roughly in alphabetical order